tensorwaves/physics/helicityformalism/kinematics.py

```diff
@@ -1,16 +1,262 @@
+r"""Kinematic based calculations for the helicity formalism.
 
-class Kinematics():
-  def __init__(self):
-    pass
+It's responsibilities are defined by the interface
+:class:`.interfaces.Kinematics`.
 
-  def register_invariant_mass(self):
-    pass
+Here, the main responsibility is the conversion of general kinematic
+information of a reaction to helicity formalism specific quantities
 
-  def register_helicity_angles(self):
-    pass
+:math:`(s, \theta, \phi)`
 
-  def register_subsystem(self):
-    pass
+The basic building blocks are the :class:`~HelicityKinematics` and
+:class:`~SubSystem`.
+"""
 
-  def convert(events):
-    pass
\ No newline at end of file
+import logging
+from typing import Union
+
+import amplitf.kinematics as tfa_kin
+
+import numpy as np
+
+from tensorwaves.interfaces import Kinematics
+
+
+class SubSystem():
+    """Represents a part of a decay chain.
+
+    A SubSystem resembles a decaying state and its ingoing and outgoing state.
+    It is uniquely defined by
+
+    * :attr:`final_states`
+    * :attr:`recoil_state`
+    * :attr:`parent_recoil_state`
+    """
+
+    def __init__(self, final_states, recoil_state, parent_recoil_state):
+        """Fully initializes the :class:`~SubSystem`.
+
+        Args:
+            final_states: `tuple` of `tuple` s containing unique ids.
```

Comment:
The unique IDs come from ComPWA, but I wonder if there is not an other way to do this, for instance with the original particle names as strings (or their PIDs), but an index added to duplicate entries. In a way also related to the discusion about `EventCollection` (https://github.com/ComPWA/tensorwaves/issues/23).

Reply:
I mention it also because I see problems coming up with labels like these again
https://github.com/ComPWA/tensorwaves/pull/39/files#diff-1d2eb2f144ec58acdde4c21d96eede1bR138

Reply:
Good point. Actually since its python now, there is actually no type defined on this unique id. It just has to uniquely identify a particle from the event collection later on.
Regarding this fact and that this would also require to change the model recipe file description (refer to names instead of ints), this deserves a separate issue and discussion.

Reply:
Maybe better to decide on this while refactoring the expert system? https://github.com/ComPWA/tensorwaves/issues/13

Reply:
Yeah, and like I tried to say above, this module is actually not limiting the functionality here. Its more the way events are generated and what mapping is handed to the kinematics instance. Btw you should resolve this if you are happy with it

Reply:
Yeah sure, it's just that the question mark left it open